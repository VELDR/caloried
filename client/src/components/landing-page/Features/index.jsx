import PropTypes from 'prop-types';
import { FormattedMessage, injectIntl } from 'react-intl';
import { EditCalendar, PieChart, QueryStats, RestaurantMenu, Scale } from '@mui/icons-material';
import FeatureCard from '@components/landing-page/FeatureCard';

import classes from './style.module.scss';

const Features = ({ intl: { formatMessage } }) => (
  <div className={classes.featuresContainer} data-testid="features">
    <div className={classes.featuresTitle}>
      <FormattedMessage id="app_features" />
    </div>
    <div className={classes.features}>
      <FeatureCard
        icon={<PieChart />}
        title={formatMessage({ id: 'app_nutrition_tracking' })}
        subtitle={formatMessage({ id: 'app_nutrition_tracking_description' })}
      />
      <FeatureCard
        icon={<EditCalendar />}
        title={formatMessage({ id: 'app_meal_planning' })}
        subtitle={formatMessage({ id: 'app_meal_planning_description' })}
      />
      <FeatureCard
        icon={<Scale />}
        title={formatMessage({ id: 'app_weight_tracking' })}
        subtitle={formatMessage({ id: 'app_weight_tracking_description' })}
      />
      <FeatureCard
        icon={<RestaurantMenu />}
        title={formatMessage({ id: 'app_food_database' })}
        subtitle={formatMessage({ id: 'app_food_database_description' })}
      />
      <FeatureCard
        icon={<QueryStats />}
        title={formatMessage({ id: 'app_dashboard_analytics' })}
        subtitle={formatMessage({ id: 'app_dashboard_analytics_description' })}
      />
    </div>
  </div>
);

Features.propTypes = {
  intl: PropTypes.object,
};

export default injectIntl(Features);
