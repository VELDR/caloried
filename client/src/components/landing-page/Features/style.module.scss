@import '@styles/base';

.featuresContainer {
  @include flex-center-column(2rem);
  background-color: var(--color-bg-secondary);
  padding: 2rem;
  .featuresTitle {
    font-size: $font-size-huge;
    font-weight: 700;
  }

  .features {
    @include flex-center(1rem);
    flex-wrap: wrap;
  }
}
