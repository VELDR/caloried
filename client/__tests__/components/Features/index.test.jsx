import Features from '@components/landing-page/Features';
import { render } from '@utils/testHelper';

let wrapper;

const mockProps = {
  formatMessage: jest.fn(),
};

beforeEach(() => {
  wrapper = render(<Features {...mockProps} />);
});

describe('Features Component', () => {
  it('should render the component correctly', () => {
    const { getByTestId } = wrapper;

    expect(getByTestId('features')).toBeInTheDocument();
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
