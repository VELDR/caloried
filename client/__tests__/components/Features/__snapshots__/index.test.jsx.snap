// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features Component should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="featuresContainer"
        data-testid="features"
      >
        <div
          class="featuresTitle"
        >
          Fitur
        </div>
        <div
          class="features"
        >
          <div
            class="card"
            data-testid="feature-card"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="PieChartIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 2v20c-5.07-.5-9-4.79-9-10s3.93-9.5 9-10zm2.03 0v8.99H22c-.47-4.74-4.24-8.52-8.97-8.99zm0 11.01V22c4.74-.47 8.5-4.25 8.97-8.99h-8.97z"
              />
            </svg>
            <div
              class="card__description"
            >
              <div
                class="title"
              >
                Pemantauan Nutrisi
              </div>
              <div
                class="subtitle"
              >
                Analisis asupan diet yang terperinci.
              </div>
            </div>
          </div>
          <div
            class="card"
            data-testid="feature-card"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="EditCalendarIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 22H5c-1.11 0-2-.9-2-2l.01-14c0-1.1.88-2 1.99-2h1V2h2v2h8V2h2v2h1c1.1 0 2 .9 2 2v6h-2v-2H5v10h7v2zm10.13-5.01.71-.71c.39-.39.39-1.02 0-1.41l-.71-.71a.9959.9959 0 0 0-1.41 0l-.71.71 2.12 2.12zm-.71.71-5.3 5.3H14v-2.12l5.3-5.3 2.12 2.12z"
              />
            </svg>
            <div
              class="card__description"
            >
              <div
                class="title"
              >
                Penyusunan Menu
              </div>
              <div
                class="subtitle"
              >
                Rancang menu mingguan dengan praktis.
              </div>
            </div>
          </div>
          <div
            class="card"
            data-testid="feature-card"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ScaleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M14 11V8c4.56-.58 8-3.1 8-6H2c0 2.9 3.44 5.42 8 6v3c-3.68.73-8 3.61-8 11h6v-2H4.13c.93-6.83 6.65-7.2 7.87-7.2s6.94.37 7.87 7.2H16v2h6c0-7.39-4.32-10.27-8-11zm-2 11c-1.1 0-2-.9-2-2 0-.55.22-1.05.59-1.41C11.39 17.79 16 16 16 16s-1.79 4.61-2.59 5.41c-.36.37-.86.59-1.41.59z"
              />
            </svg>
            <div
              class="card__description"
            >
              <div
                class="title"
              >
                Pemantauan Berat Badan
              </div>
              <div
                class="subtitle"
              >
                Pantau dan visualisasikan progres berat badan.
              </div>
            </div>
          </div>
          <div
            class="card"
            data-testid="feature-card"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="RestaurantMenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m8.1 13.34 2.83-2.83L3.91 3.5c-1.56 1.56-1.56 4.09 0 5.66l4.19 4.18zm6.78-1.81c1.53.71 3.68.21 5.27-1.38 1.91-1.91 2.28-4.65.81-6.12-1.46-1.46-4.2-1.1-6.12.81-1.59 1.59-2.09 3.74-1.38 5.27L3.7 19.87l1.41 1.41L12 14.41l6.88 6.88 1.41-1.41L13.41 13l1.47-1.47z"
              />
            </svg>
            <div
              class="card__description"
            >
              <div
                class="title"
              >
                Jutaan Data Makanan
              </div>
              <div
                class="subtitle"
              >
                Temukan informasi nutrisi berbagai makanan.
              </div>
            </div>
          </div>
          <div
            class="card"
            data-testid="feature-card"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="QueryStatsIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.88 18.47c.44-.7.7-1.51.7-2.39 0-2.49-2.01-4.5-4.5-4.5s-4.5 2.01-4.5 4.5 2.01 4.5 4.49 4.5c.88 0 1.7-.26 2.39-.7L21.58 23 23 21.58l-3.12-3.11zm-3.8.11c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5zm-.36-8.5c-.74.02-1.45.18-2.1.45l-.55-.83-3.8 6.18-3.01-3.52-3.63 5.81L1 17l5-8 3 3.5L13 6l2.72 4.08zm2.59.5c-.64-.28-1.33-.45-2.05-.49L21.38 2 23 3.18l-4.69 7.4z"
              />
            </svg>
            <div
              class="card__description"
            >
              <div
                class="title"
              >
                Analitik Dasbor 
              </div>
              <div
                class="subtitle"
              >
                Wawasan tentang pola makan Anda.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="featuresContainer"
      data-testid="features"
    >
      <div
        class="featuresTitle"
      >
        Fitur
      </div>
      <div
        class="features"
      >
        <div
          class="card"
          data-testid="feature-card"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PieChartIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 2v20c-5.07-.5-9-4.79-9-10s3.93-9.5 9-10zm2.03 0v8.99H22c-.47-4.74-4.24-8.52-8.97-8.99zm0 11.01V22c4.74-.47 8.5-4.25 8.97-8.99h-8.97z"
            />
          </svg>
          <div
            class="card__description"
          >
            <div
              class="title"
            >
              Pemantauan Nutrisi
            </div>
            <div
              class="subtitle"
            >
              Analisis asupan diet yang terperinci.
            </div>
          </div>
        </div>
        <div
          class="card"
          data-testid="feature-card"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="EditCalendarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 22H5c-1.11 0-2-.9-2-2l.01-14c0-1.1.88-2 1.99-2h1V2h2v2h8V2h2v2h1c1.1 0 2 .9 2 2v6h-2v-2H5v10h7v2zm10.13-5.01.71-.71c.39-.39.39-1.02 0-1.41l-.71-.71a.9959.9959 0 0 0-1.41 0l-.71.71 2.12 2.12zm-.71.71-5.3 5.3H14v-2.12l5.3-5.3 2.12 2.12z"
            />
          </svg>
          <div
            class="card__description"
          >
            <div
              class="title"
            >
              Penyusunan Menu
            </div>
            <div
              class="subtitle"
            >
              Rancang menu mingguan dengan praktis.
            </div>
          </div>
        </div>
        <div
          class="card"
          data-testid="feature-card"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ScaleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14 11V8c4.56-.58 8-3.1 8-6H2c0 2.9 3.44 5.42 8 6v3c-3.68.73-8 3.61-8 11h6v-2H4.13c.93-6.83 6.65-7.2 7.87-7.2s6.94.37 7.87 7.2H16v2h6c0-7.39-4.32-10.27-8-11zm-2 11c-1.1 0-2-.9-2-2 0-.55.22-1.05.59-1.41C11.39 17.79 16 16 16 16s-1.79 4.61-2.59 5.41c-.36.37-.86.59-1.41.59z"
            />
          </svg>
          <div
            class="card__description"
          >
            <div
              class="title"
            >
              Pemantauan Berat Badan
            </div>
            <div
              class="subtitle"
            >
              Pantau dan visualisasikan progres berat badan.
            </div>
          </div>
        </div>
        <div
          class="card"
          data-testid="feature-card"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="RestaurantMenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m8.1 13.34 2.83-2.83L3.91 3.5c-1.56 1.56-1.56 4.09 0 5.66l4.19 4.18zm6.78-1.81c1.53.71 3.68.21 5.27-1.38 1.91-1.91 2.28-4.65.81-6.12-1.46-1.46-4.2-1.1-6.12.81-1.59 1.59-2.09 3.74-1.38 5.27L3.7 19.87l1.41 1.41L12 14.41l6.88 6.88 1.41-1.41L13.41 13l1.47-1.47z"
            />
          </svg>
          <div
            class="card__description"
          >
            <div
              class="title"
            >
              Jutaan Data Makanan
            </div>
            <div
              class="subtitle"
            >
              Temukan informasi nutrisi berbagai makanan.
            </div>
          </div>
        </div>
        <div
          class="card"
          data-testid="feature-card"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="QueryStatsIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.88 18.47c.44-.7.7-1.51.7-2.39 0-2.49-2.01-4.5-4.5-4.5s-4.5 2.01-4.5 4.5 2.01 4.5 4.49 4.5c.88 0 1.7-.26 2.39-.7L21.58 23 23 21.58l-3.12-3.11zm-3.8.11c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5zm-.36-8.5c-.74.02-1.45.18-2.1.45l-.55-.83-3.8 6.18-3.01-3.52-3.63 5.81L1 17l5-8 3 3.5L13 6l2.72 4.08zm2.59.5c-.64-.28-1.33-.45-2.05-.49L21.38 2 23 3.18l-4.69 7.4z"
            />
          </svg>
          <div
            class="card__description"
          >
            <div
              class="title"
            >
              Analitik Dasbor 
            </div>
            <div
              class="subtitle"
            >
              Wawasan tentang pola makan Anda.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
